textract-lambda-sam-python/README.md

```diff
@@ -0,0 +1,70 @@
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the pattern directory:
+    ```
+    cd textract-lambda-sam-python
+    ```
+3. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam deploy --guided
+    ```
+4. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Allow SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+5. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+
+- This pattern is designed to create all services required to run this workflow.
+- The workflow begins with an Amazon S3 bucket. 
+- When an object is created within the S3 bucket, it sends an event to a AWS Lambda function.
+- This Lambda function invokes Amazon Textracts's [DetectDocumentText](https://docs.aws.amazon.com/textract/latest/dg/API_DetectDocumentText.html) function, which synchronously analyses the newly stored file.
+- Once this task is complete, the Lambda function stores the results in an Amazon DynamoDB table.
+
+## Testing
+1. Upload a test image or pdf file which contains some text to the Amazon S3 bucket created during the deployment step. You can either do this via the console, or by running the following command. Replace `<your_file>` with your image or pdf file name. Replace `<S3BucketName>` with the name of the S3 bucket generated from the AWS SAM deployment outputs.
+    ```
+    aws s3 cp <your_file> s3://<S3BucketName>
+    ```
+3. Wait for the lambda-start-detect-document-text-textract Lambda function to complete, then retrieve the record output from the DynamoDB table. Replace `<DDBTableName>` with the name of the DynamoDB table generated by the AWS SAM deployment output, in this case, it will be `TextractResultsTable`. You can use the following command:
+    ```
+    aws dynamodb scan --table-name <DDBTableName>
+    ```
+    For example: aws dynamodb scan --table-name TextractResultsTable
+4. There should now be a newly uploaded record within DynamoDB displayed, depending on the file, the key 'DetectedText' will either be empty, or contain values indicating what was found in the uploaded file.
+
+## Cleanup
+ 
+1. Delete the stack
+    ```
+    sam delete
+    ```
+
+⚠️ **IMPORTANT** - The above command `sam delete` does not delete the **Amazon S3 bucket** if there are still objects stored within it. Please [empty the bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/empty-bucket.html) before running the above command.
```

Comment:
I would add a DeletionPolicy to the bucket and have instructions to empty the bucket before running sam delete.